cocotb/scheduler.py

```diff
@@ -709,18 +707,12 @@ def schedule(self, coroutine, trigger=None):
 
         try:
             result = self._trigger_from_any(result)
-        except TypeError:
-            msg = ("Coroutine %s yielded something the scheduler can't handle"
```

Comment:
No need to include the name of the coroutine in the message any more, it appears in the stack trace, along with the line that performed the bad yield.